UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -287,13 +284,13 @@ private void PeriodicUpdate()
 		{
 			if (IsDead == false)
 			{
-				for (int i = BodyPartList.Count - 1; i >= 0; i--)
+				foreach (var bodyPart in BodyPartList)
 				{
-					BodyPartList[i].ImplantPeriodicUpdate();
+					bodyPart.ImplantPeriodicUpdate();
 				}
```

Comment:
Are you sure these for loops arent needed due to changing the BodyPartList and so needs to be looped backwards?

Reply:
yes